pkg/scale/decode.go

```diff
@@ -60,26 +60,23 @@ func indirect(dstv reflect.Value) (elem reflect.Value) {
 func Unmarshal(data []byte, dst interface{}) (err error) {
 	dstv := reflect.ValueOf(dst)
 	if dstv.Kind() != reflect.Ptr || dstv.IsNil() {
-		err = fmt.Errorf("unsupported dst: %T, must be a pointer to a destination", dst)
+		err = fmt.Errorf("%w: %T", ErrUnsupportedDestination, dst)
 		return
 	}
 
 	elem := indirect(dstv)
-	if err != nil {
-		return
-	}
 
 	buf := &bytes.Buffer{}
 	ds := decodeState{}
 	_, err = buf.Write(data)
 	if err != nil {
-		return
+		return err
 	}
 	ds.Reader = buf
 
 	err = ds.unmarshal(elem)
 	if err != nil {
-		return
+		return fmt.Errorf("unmarshalling: %w", err)
```

Comment:
nit having more thoughts on that, since the function is called `Unmarshal`, I guess we can just `return err` in that case since it would just repeat `unmarshaling: unmarshaling: actual error`.

```suggestion
		return err
```